_posts/2017-07-06-hitchhikers-guide-to-cp.md

```diff
@@ -0,0 +1,308 @@
+
+    +   Short contests are best for improving speed and accuracy.
+
++   **[Codeforces](http://codeforces.com/) :**
+
+    +   4-5 Shorts contests(2 hrs) every month.
+
+    +   Very effective in improving speed and accuracy.
+
+    +   Less time, more problems and more participants makes it tougher and more interesting than Codechef.
+    
+---
+
+# Getting Started
+
+In order to solve a problem, you first read the problem, code the solution and submit the solution online. A verdict is shown from which you can see how your code performed against the test cases ran by the online judge.
+
+## Problem Statement:
+
+Statement starts with few paragraphs explaining the problem. Read it thoroughly until you understand what is to be solved.
+
+After that, there's *Input and output* format explaining how input and output is formatted. NEVER output anything else than what is asked. For example, do not print stuff like "Enter a number: ".
+
+Then, *Constraints* are given which will give you the idea of how big the input is and whether your approach will work or not. 
+Constraints include --
++   **Time limit :** Given in *seconds*. Time less than which your code should calculate and print the output.
+
++   **Memory limit :** Given in *MB*. Maximum amount of memory your code should use.
+
++   **Input Size :** Maximum possible input size, ie, input bigger than this won't be there in test cases.
+
+Then, there will be *Sample Input* and *Sample Output* to give you an idea of I/O formats and to give you a better understanding of the problem.
+
+Lastly, there will be *Explanation*(might be absent) which explains how sample output is calculated.
+
+## Solving the problem:
+
+After reading and understanding the problem, a solution is to be coded for it. A given problem can have many possible solutions but you have to find that solution which comes under the constraints. In order to see if your code fulfills the time and memory limits, you need to do an *Order Analysis* of your solution.
+
+**Order Analysis :** 
+Order of a solution is a measure of it's runtime or number of operations as a function of input size. Let's take an example, Suppose we have to find the maximum among N positive numbers.
+
+```
+int ans = -1,x;
+for(int i=0;i<N;i++) {
+    scanf(“%d”,&x);
+    ans = max(ans, x);
+}
+```
+
+N times checking i < N.
+
+N times checking max of ans and x.
+
+N times i++.
+
+N times scanf.
+
+Hence, total number of operations is $$4 * N$$. We only care about dependency on N. Therefore, order is $$O(N)$$.
+
+[CLRS](https://www.google.co.in/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0ahUKEwju8crC0e_UAhWHN48KHTAVCdoQFggjMAA&url=https%3A%2F%2Fweb.njit.edu%2F~wl256%2Fdownload%2Fcs610%2FIntroduction-to-algorithm-3rdEdition.pdf&usg=AFQjCNGJTrGFQdXhlY7ItJk2pCGX9V5S9Q) is having detailed theoretical explanation on Order Analysis.
```

Comment:
Ideally, I would not want to directly promote piracy of books by linking the book here. Maybe change to a wikipedia link?